src/leo-client-app/src/components/operator-management/EditUserRoleModal.tsx

```diff
@@ -0,0 +1,107 @@
+import { updateOperatorRole } from "@/constants/api";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  MenuItem,
+  Select,
+  Stack,
+} from "@mui/material";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+
+type Props = {
+  open: boolean;
+  userData: any;
+  handleClose: () => void;
+};
+
+enum UserRole {
+  "OPERATOR" = "OPERATOR",
+  "ADMIN" = "ADMIN",
+}
+
+const EditUserRoleModal = ({ open, userData, handleClose }: Props) => {
+  const queryClient = useQueryClient();
+
+  const [userRole, setUserRole] = useState<UserRole>(
+    userData?.role ?? UserRole.OPERATOR
```

Comment:
getting google flashbacks from this, can we avoid userData?.role (or type it out) just because it can get confusing looking at it 

up to you thought

Reply:
whhoops